components/expenses/ExpenseForm.js

```diff
@@ -117,7 +118,10 @@ export const prepareExpenseForSubmit = expenseData => {
       ? pick(expenseData.payee, ['name', 'email', 'legalName', 'organization', 'newsletterOptIn'])
       : { [payeeIdField]: expenseData.payee.id };
 
-  const payeeLocation = isInvoice ? pick(expenseData.payeeLocation, ['address', 'country', 'structured']) : null;
+  const payeeLocation =
+    isInvoice || ![CollectiveType.COLLECTIVE, CollectiveType.ORGANIZATION].includes(expenseData.payee.type)
```

Comment:
We shouldn't stop requiring addresses from organizations, only collectives don't need them because their legal address is the one from their fiscal host.

Also, can we reduce the duplication by using a small helper? `checkRequiresAddress` maybe?

Reply:
Yes I've extracted this to a small helper. Let me know how it looks now. 👍🏽